### PR TITLE
chore(backend): move v1 api to v1beta1 package and folder

### PR DESCRIPTION
**Pull request by Linchin**
**Description of your changes:**
* Moved v1 api related files to the `v1beta1` folder
* Changed the proto package name from `api` into `v1beta1`
* Updated scripts to generate Goalng package and Python package
* Also moved the auto generated Golang package and Python package into `v1beta1` folder
* Updated `README.md` on instructions to generate Golang and Python packages
* Refactored backend to use the generated `v1beta1` package

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Review by gkcalat — CHANGES_REQUESTED** [review]
Hi @Linchin! It seems that you also need to adjust the tests. For example, [here](https://github.com/kubeflow/pipelines/blob/master/test/sample-test/Dockerfile#L14).

**Comment by chensun** [comment]
> All the generated files under `go_http_client` are deleted. Forgot to add the new files, or the script to generate has some bug?

Same for `python_http_client`

**Comment by Linchin** [comment]
> All the generated files under `go_http_client` are deleted. Forgot to add the new files, or the script to generate has some bug?

They are all in `v1beta1` folder.

File: backend/api/README.md [hunk 1]
```diff
@@ -9,23 +9,35 @@ Tools needed:
...
+
+Note: `./v1beta1/swagger/pipeline.upload.swagger.json` is manually created, while the rest `./v1beta1/swagger/*.swagger.json` are auto generated.
 
 ## Auto-generation of Python client
 
+Add the API version after the command to generate the corresponding Python client. We use `v1beta1` as an example here:
+
 ```bash
-./build_kfp_server_api_python_package.sh
+./build_kfp_server_api_python_package.sh v1beta1
```
Comment (gkcalat) [inline backend/api/README.md:37]: NIT: Should we use `${API_VERSION}` everywhere in the readme?

**Comment by chensun** [comment]
> > All the generated files under `go_http_client` are deleted. Forgot to add the new files, or the script to generate has some bug?
> 
> They are all in `v1beta1` folder.

My bad, I thought it would show as moved.

File: backend/api/README.md (hunk 1 above)
Comment (chensun) [reply backend/api/README.md:37]: +1

Comment (Linchin) [reply backend/api/README.md:37]: I'm just using `v1beta1` as an example for the user here. The user needs to specify the version of the api to generate.

File: backend/api/README.md [hunk 2]
```diff
@@ -9,23 +9,35 @@ Tools needed:
 
 ## Auto-generation of Go client and swagger definitions
 
+Set environment variable `API_VERSION` to the version that you want to generate. We use `v1beta1` as example here.
+
+```bash
+API_VERSION="v1beta1" make generate
```
Comment (chensun) [inline backend/api/README.md:15]: remove `make generate`

File: backend/api/README.md [hunk 3]
```diff
@@ -9,23 +9,35 @@ Tools needed:
...
+```
+
+For local developments, use 
 ```bash
-make generate
+API_VERSION="v1beta1" make generate-from-scratch
```
Comment (chensun) [inline backend/api/README.md:20]: remove `API_VERSION="v1beta1"` as it's defined already from above.

Comment (chensun) [reply backend/api/README.md:20]: Also why change `make generate` to `make generate-from-scratch`?

File: backend/api/README.md (hunk 1 above)
Comment (chensun) [reply backend/api/README.md:37]: That is already specified via `API_VERSION`, I think the suggestion here is just to reuse it across the doc.

**Review by gkcalat — APPROVED** [review]

File: backend/api/README.md (hunk 3 above)
Comment (Linchin) [reply backend/api/README.md:20]: I think the wording here is a bit confusing. There are two ways to generate the api package: using `make generate`, which uses a pre-built api-generator image; or you can use `make generate-from-scratch`, which starts by build the api-generator image locally. I updated the wording to make it clearer.

Comment (Linchin) [reply backend/api/README.md:21]: And `API_VERSION="v1beta1"` is an integral part of the command.

Comment (chensun) [reply backend/api/README.md:21]: I don't get why you need to build the image for local development, you're not changing anything related to the image. Did you hit any specific error with `make generate` that leads you to rebuilding the image? If the hosted image is broken, we need to update it.

All the nitpicks combined was to simply reuse `API_VERSION` across the doc. 
The doc would be something like this:
## Set version
```bash
export API_VERSION="v1beta1"
```
## Generate go client
```bash
make generate
```
Code will be generated in
* `./"${API_VERSION}"/go_client`
* ...
## Generate python client
```bash
./build_kfp_server_api_python_package.sh
```
Code will be generated in `./"${API_VERSION}"/python_http_client`.

(need changes to `build_kfp_server_api_python_package.sh` so it reads `$API_VERSION`)

File: backend/api/README.md [hunk 4]
```diff
@@ -9,23 +9,35 @@ Tools needed:
...
+
 ```bash
-./build_kfp_server_api_python_package.sh
+./build_kfp_server_api_python_package.sh v1beta1
 ```
 
-Code will be generated in `./python_http_client`.
+Code will be generated in `./v1beta1/python_http_client`.
 
 ## Auto-generation of API reference documentation
```
Comment (chensun) [inline backend/api/README.md:42]: I think this section needs update as well since the swagger files are moved.

File: backend/api/README.md (hunk 3 above)
Comment (Linchin) [reply backend/api/README.md:21]: In this PR the image is not changed, but in future developments, if the Dockerfile is changed, the image need to be rebuilt. This is why there are two options to generate the packages.

Comment (Linchin) [reply backend/api/README.md:21]: As to set version as an environment variable, I will update the doc accordingly.

Comment (Linchin) [reply backend/api/README.md]: The command `make-generate-from-scratch` has always been in the `Makefile`. I just added more lines in the readme so it's more convenient for future devs.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**